default_handler.go

```diff
@@ -38,12 +38,29 @@ type defaultHandler struct {
 	defaultIntf map[string]*exportedIntf
 }
 
-func (h *defaultHandler) PathExists(path ObjectPath) bool {
-	_, ok := h.objects[path]
-	return ok
+func (h *defaultHandler) GetExportedObject(path ObjectPath) (*exportedObj, bool) {
+	h.RLock()
+	defer h.RUnlock()
+	obj, ok := h.objects[path]
+	return obj, ok
+}
+
+// GetAnExportedObject returns an exportedObj for an specific ObjectPath
+// A new exportedObj is creaated if none existed for ObjectPath
+func (h *defaultHandler) GetAnExportedObject(path ObjectPath) *exportedObj {
+	h.RLock()
+	defer h.RUnlock()
```

Comment:
This needs to be a write lock then. The race detector picked this up as well: https://github.com/godbus/dbus/actions/runs/4797626686/jobs/9033883115?pr=340#step:6:243

Reply:
So true. Fixed!